pkg/connector/handlematrix.go

```diff
@@ -344,3 +346,107 @@ func (s *SignalClient) HandleMatrixRoomTopic(ctx context.Context, msg *bridgev2.
 		ModifyDescription: &msg.Content.Topic,
 	}, nil)
 }
+
+func (s *SignalClient) HandleMatrixMembership(ctx context.Context, msg *bridgev2.MatrixMembershipChange) (bool, error) {
+	var targetIntent bridgev2.MatrixAPI
+	var targetSignalID uuid.UUID
+	var err error
+	if msg.TargetGhost != nil {
+		targetIntent = msg.TargetGhost.Intent
+		targetSignalID, err = signalid.ParseUserID(msg.TargetGhost.ID)
+		if err != nil {
+			return false, fmt.Errorf("failed to parse target ghost signal id: %w", err)
+		}
+	} else if msg.TargetUserLogin != nil {
+		targetSignalID, err = signalid.ParseUserLoginID(msg.TargetUserLogin.ID)
+		if err != nil {
+			return false, fmt.Errorf("failed to parse target user signal id: %w", err)
+		}
+		targetIntent = msg.TargetUserLogin.User.DoublePuppet(ctx)
+		if targetIntent == nil {
+			ghost, err := s.Main.Bridge.GetGhostByID(ctx, networkid.UserID(msg.TargetUserLogin.ID))
+			if err != nil {
+				return false, fmt.Errorf("failed to get ghost for user: %w", err)
+			}
+			targetIntent = ghost.Intent
+		}
+	}
+	gc := &signalmeow.GroupChange{}
+	role := signalmeow.GroupMember_DEFAULT
+	toMembership := msg.Type.GetTo()
+	if toMembership == event.MembershipBan {
+		gc.AddBannedMembers = []*signalmeow.BannedMember{{
+			ServiceID: libsignalgo.NewACIServiceID(targetSignalID),
```

Comment:
Why is this case separately from the switch 🤔

Reply:
I separated Bans into different membership transitions, depending on what the previous membership is. Those remove the user from the group, remove an invite etc. I could add the BannedMember each time, but I figured it would be easier to only do that once.

Reply:
You can have a `case` with multiple values (`case bridgev2.Ban, bridgev2.BanJoined, bridgev2.BanInvited: ...`)

Reply:
I know that, but I would then have to separate out the different from memberships at that point - I don't see the advantage over what I'm currently doing.

Reply:
The advantage is having a single switch statement that covers every membership change, which is cleaner and allows having a `default` block for unsupported changes